File: Lönn/entities/trampoline.lua
local trampoline = {}

trampoline.name = "GameHelper/Trampoline"
trampoline.depth = 8998
trampoline.texture = "objects/GameHelper/trampoline/idle"
trampoline.placements = {
    name = "trampoline",
    data = {
        speedBoostX = 60.0,
        speedBoostY = 60.0,
        facingUpLeft = true,
        refillDash = true,
        oneUse = false
    }
}

--function trampoline.justification(room, entity)
--    return entity.facingUpLeft and {1.0, 0.5} or {0.0, 0.5}
--end

function trampoline.rotation(room, entity)
    return entity.facingUpLeft and 0 or math.pi / 2
end

return trampoline

File: Ahorn/entities/trampoline.jl
module GameHelperTrampoline

using ..Ahorn, Maple

@mapdef Entity "GameHelper/Trampoline" Trampoline(
    x::Integer, y::Integer, speedBoost::Number=40.0, facingUpLeft::Bool=true, refillDash::Bool=true, oneUse::Bool=false
)

const placements = Ahorn.PlacementDict(
    "Trampoline (GameHelper)" => Ahorn.EntityPlacement(
        Trampoline
    )
)

function Ahorn.selection(entity::Trampoline)
    x, y = Ahorn.position(entity)
    sprite = "objects/GameHelper/trampoline/idle.png"
    return Ahorn.getSpriteRectangle(sprite, x, y)
end

function Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::Trampoline, room::Maple.Room)
theSprite = "objects/GameHelper/trampoline/idle.png"
sprite = theSprite
Ahorn.drawSprite(ctx, sprite, 0, 0)
end

end